recipes/renode-cli/helpers/renode_build_with_dotnet.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+
+set -uex
+
+framework_version=$1
+shift 1
+
+OUTPUT_DIRECTORY="${SRC_DIR}/output"
+
+CONFIGURATION="Release"
+BUILD_PLATFORM="Any CPU"
+HEADLESS=true
+NET=true
+TFM="net${framework_version}"
+HOST_ARCH="i386"
+# Common cmake flags
+CMAKE_COMMON=""
```

Comment:
Please reduce the usage of variables in these helper scripts only to those values which need to differ between platforms.  It makes the code logic very difficult to follow.  Use line breaks with `\ ` if the line is too long.  If you have entire arguments that are only needed on one platform, then just use a simple string rather than an array.  If we can simplify the logic here enough, we can just move this code into the main `build.sh` script.

Reply:
Right, a lot of it is again from their original design, I carved out their build script (if I remember correctly). now that you semi-approve the overall structure, I probably can simplify a bunch as you suggested

Reply:
Indeed, now that we know what it's doing we can simplify a lot more.  I'll check in once you've had a chance to do more cleanup - while this recipe will never be as simple as most other .NET recipes, I'm fairly sure we can get it to the point that it's pretty straight forward to follow even if it's a bit longer.